src/tbp/monty/frameworks/config_utils/make_dataset_configs.py

```diff
@@ -1102,3 +1104,276 @@ class FiveLMMountConfig:
     zooms: List[float] = field(
         default_factory=lambda: [10.0, 10.0, 10.0, 10.0, 10.0, 1.0]
     )
+
+
+"""
+Utilities for generating arbitrary multi-LM dataset args.
+"""
+
+
+def make_multi_lm_sensor_positions(
+    n_sms: int,
+    delta: Number = 0.01,
+    order_by: str = "distance",
+    add_view_finder: bool = True,
+) -> np.ndarray:
+    """Generate sensor positions based on a 2D grid.
+
+    Create mounting positions for an arbitrary number of sensor module, where the
+    sensor modules lie along an imaginary grid on the xy plane (and z = 0). Sensor
+    module 0 is always centered at (0, 0, 0), and all other sensors are clustered
+    around it. The method for selecting which grid points around the center to assign
+    to sensor modules is determined by the `order_by` argument (see below).
+
+    By default, `n_sms + 1` positions are returned; the first `n_sms` positions are for
+    regular sensor modules, and an additional position is appended to accommodate
+    a view finder. The view finder position (if used) is also centered on (0, 0, 0).
+
+    Args:
+        n_sms (int): Number of sensor modules.
+        delta (number): The grid spacing length. By default, sensor modules will be
+            placed every centimeter (units are in meters).
+        order_by (str, optional): How to select points on the grid that will contain
+            sensor modules.
+             - "spiral": sensor modules are numbered along a counter-clockwise spiral
+                spreading outwards from the center.
+             - "distance": sensor modules are ordered by their distance from the center.
+                This can result in a more jagged pattern along the edges but
+                results in sensor modules generally more packed towards the center.
+                Positions that are equidistant from the center are ordered
+                counterclockwise starting at 3 o'clock.
+        add_view_finder (bool, optional): Whether to include an extra position module
+            at the origin to serve as a view finder. Defaults to `True`.
+
+    Returns:
+        np.ndarray: A 2D array of sensor positions where each row has positions
+            (x, y, z). If `add_view_finder` is True, the array has `n_sms + 1` rows,
+            where the last row corresponds to the view finder's position and is
+            identical to row 0. Otherwise, the array has `n_sms` rows. row 0 is
+            always centered at (0, 0, 0), and all other rows are offset relative to it.
+
+    """
+    assert n_sms > 0, "n_lms must be greater than 0"
+    assert delta > 0, "delta must be greater than 0"
+    assert order_by in ["spiral", "distance"], "order_by must be 'spiral' or 'distance'"
+
+    # Find smallest square grid size that can fit n_lms with odd-length sides.
+    n = 1
+    while n_sms > n**2:
+        n += 2
+
+    # Make coordinate grids, where the center is (0, 0).
+    pts = np.arange(-n // 2 + 1, n // 2 + 1)
```

Comment:
also here I would not abbreviate. i assume it stands for "points"?

Reply:
Yep, changed to `points`.